### PR TITLE
Refactor TransferAsset acceptance test

### PR DESCRIPTION
**Pull request by l4l**
### Description of the Change

- Add IntegrationTestFramework::sendTxAwait
- Reuse AcceptanceFixture primitive in TA test
- Move out primitive methods in fixture
- Fix WithOnlyCanTransferPerm/WithOnlyCanReceivePerm tests

Also, [I've tried](http://termbin.com/ega0) to remove `Role::kAddAssetQty` from the first user but it seems to have some bug. Feel free to review/comment it as well

### Benefits

Lesser simpler code -> lesser bugs

### Possible Drawbacks 

None?

### Usage Examples or Tests

`make transfer_asset_test && test_bin/transfer_asset_test`

### CODE REVIEW

File: test/integration/acceptance/transfer_asset_test.cpp [hunk 1]
```diff
@@ -19,159 +19,103 @@ using namespace shared_model;
...
 
 /**
- * @given some user with only can_transfer permission
+ * @given pair of users
+ *        AND the first user without can_transfer permission
  * @when execute tx with TransferAsset command
  * @then there is an empty proposal
  */
-TEST_F(TransferAsset, WithOnlyCanTransferPerm) {
+TEST_F(TransferAsset, DISABLED_WithOnlyCanTransferPerm) {
```
Comment (igor-egorov) [inline test/integration/acceptance/transfer_asset_test.cpp]: Please specify in a comment the reason why the test is disabled. If the test can be enabled only after resolution of some other issue, denote it as a todo comment with Jira task id.

File: test/integration/acceptance/transfer_asset_test.cpp [hunk 2]
```diff
@@ -19,159 +19,103 @@ using namespace shared_model;
...
 
 /**
- * @given some user with only can_receive permission
+ * @given pair of users
+ *        AND the second user without can_receive permission
  * @when execute tx with TransferAsset command
  * @then there is an empty proposal
  */
-TEST_F(TransferAsset, WithOnlyCanReceivePerm) {
+TEST_F(TransferAsset, DISABLED_WithOnlyCanReceivePerm) {
```
Comment (igor-egorov) [inline test/integration/acceptance/transfer_asset_test.cpp]: Please specify in a comment the reason why the test is disabled. If the test can be enabled only after resolution of some other issue, denote it as a todo comment with Jira task id.

File: test/integration/acceptance/transfer_asset_test.cpp [hunk 3]
```diff
@@ -475,63 +358,24 @@ TEST_F(TransferAsset, BigPrecision) {
...
-      .checkBlock(
-          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
-      .sendQuery(make_query(kUser1Id), check_balance(kUser1Id, kLeft))
+      .sendTxAwait(makeFirstUser(), check(1))
+      .sendTxAwait(makeSecondUser(), check(1))
+      .sendTxAwait(create_asset, check(1))
+      .sendTxAwait(add_assets, check(1))
+      .sendTxAwait(make_transfer, check(1))
+      .sendQuery(make_query(kUserId), check_balance(kUserId, kLeft))
       .sendQuery(make_query(kUser2Id), check_balance(kUser2Id, kForTransfer))
```
Comment (igor-egorov) [inline test/integration/acceptance/transfer_asset_test.cpp]: Despite the test does not fail, the query causes postgres' error:
```
[2018-06-21 22:45:45.544989785][th:20689][info] IntegrationTestFramework send query
[2018-06-21 22:45:45.552626695][th:20689][error] PostgresWsvQuery ERROR:  invalid input value for enum grantable_perm: ""
LINE 1: ...n@test' AND account_id = 'usertwo@test' AND permission = '';
                                                                    ^
```

Comment (l4l) [reply test/integration/acceptance/transfer_asset_test.cpp]: Will be resolved with IR-1464

File: test/integration/acceptance/acceptance_fixture.cpp [hunk 4]
```diff
@@ -94,7 +94,7 @@ auto AcceptanceFixture::baseTx()
 
 auto AcceptanceFixture::baseQry()
     -> decltype(base(TestUnsignedQueryBuilder())) {
-  return base(TestUnsignedQueryBuilder());
+  return base(TestUnsignedQueryBuilder()).queryCounter(tx_counter);
```
[the remark this answers is not in the document]
Comment (l4l) [reply test/integration/acceptance/acceptance_fixture.cpp]: There's the same counter for both tx and query, I may rename it to just `counter`

**Pull request merged**